cmake/OpenCVPCHSupport.cmake

```diff
@@ -406,7 +406,9 @@ ENDMACRO(ADD_NATIVE_PRECOMPILED_HEADER)
 
 macro(ocv_add_precompiled_header_to_target the_target pch_header)
   if(PCHSupport_FOUND AND ENABLE_PRECOMPILED_HEADERS AND EXISTS "${pch_header}")
-    if(CMAKE_GENERATOR MATCHES "^Visual" OR CMAKE_GENERATOR MATCHES Xcode)
+    if(NOT CMAKE_VERSION VERSION_LESS "3.16")
+      target_precompile_headers(${the_target} PRIVATE ${pch_header})
```

Comment:
Fix is not complete: https://pullrequest.opencv.org/buildbot/builders/precommit_windows64/builds/100106
(need to adapt, GET_NATIVE_PRECOMPILED_HEADER macro)

Also please add pre-check to bypass this flow: `AND NOT OPENCV_SKIP_CMAKE_BUILTIN_PCH`